### PR TITLE
Enable debuginfod for gdb and elfutils

### PR DESCRIPTION
**Pull request by martijnthe**
- Configure elfutils to also build libdebuginfod
- Configure gdb to build with libdebuginfod

### CODE REVIEW

File: gdb-multi-arch/conda_build_config.yaml [hunk 1]
```diff
@@ -313,7 +313,7 @@ dcap:
 eclib:
   - '20230424'
 elfutils:
-  - 0.189
+  - 0.189.memfault0_libdebuginfod
```
Comment (martijnthe) [inline gdb-multi-arch/conda_build_config.yaml:316]: Not sure if this is the correct way to ensure that the `..._libdebuginfod` version gets picked up. It seems to work though?

File: gdb-multi-arch/build.sh [hunk 2]
```diff
@@ -11,6 +11,13 @@ export REAL_TARGET_PREFIX="${PREFIX}/${REAL_TARGET}"
 
 export CPPFLAGS="$CPPFLAGS -fcommon -I$PREFIX/include -Wno-constant-logical-operand"
 export CFLAGS="$CFLAGS -Wno-constant-logical-operand -Wno-format-nonliteral -Wno-self-assign"
+
+if [ `uname` == Darwin ]; then
+  EXTRA_CONFIGURE_FLAGS=""
```
Comment (noahp) [inline gdb-multi-arch/build.sh:16]: RIP

**Review by noahp — APPROVED** [review]
sweet!

**Comment by noahp** [comment]
Oh, how much did this change the package size? nb if you're building locally it's good to set these flags for outputting .conda packages instead of .tar.bz2
```bash
❯ conda config --set conda_build.pkg_format 2 && conda config --set conda_build.zstd_compression_level 19
```

File: gdb-multi-arch/conda_build_config.yaml (hunk 1 above)
Comment (noahp) [reply gdb-multi-arch/conda_build_config.yaml:316]: yup, this should work!

**Comment by martijnthe** [comment]
> > Oh, how much did this change the package size? nb if you're building locally it's good to set these flags for outputting .conda packages instead of .tar.bz2
> > ```shell
> > ❯ conda config --set conda_build.pkg_format 2 && conda config --set conda_build.zstd_compression_level 19
> > ```
> 
> The size delta for libdebuginfod.so (stub vs real thing) is about 25KB.
> 
> I'm still waiting for the gdb build: https://github.com/memfault/conda-recipes/actions/runs/6903870062 I would be very surprised if it's significantly bigger.

Double checked: there is no real difference in size in the linux binaries.

File: elfutils/meta.yaml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 package:
   name: elfutils
-  version: {{ version }}.memfault0
+  version: {{ version }}.memfault0_libdebuginfod
```
Comment (martijnthe) [inline elfutils/meta.yaml:5]: I noticed that this version ends up getting sorted as an older version, behind `...memfault0`.
I'll bump it to `.memfault1` to fix that.

**Comment by martijnthe** [comment]
@noahp could you take another look? I also applied it to the xtensa gdbs and correctly bumped all versions.

**Comment by noahp** [comment]
> @noahp could you take another look? I also applied it to the xtensa gdbs and correctly bumped all versions.

It looks good, thank you!

**Pull request merged**